Model/Api/UpdateCartItemTrait.php

```diff
@@ -130,6 +130,10 @@ protected function verifyItemData($product, $updateItem, $quoteItem, $websiteId)
             $qtyToCheck = ($updateItem['update'] == 'add')
                     ? (int)$quoteItem['quantity'] + (int)$updateItem['quantity']
                     : (int)$quoteItem['quantity'] - (int)$updateItem['quantity'];
+            
+            if ($updateItem['update'] == 'remove' && $qtyToCheck == 0) {
```

Comment:
Why do we need `$qtyToCheck == 0` check here?

Reply:
if the customer want to remove the item from cart, the update-cart api hook would have the quantity value of request equal to quote item quantity, so $qtyToCheck == 0 means the same behavior. 
Then $qtyToCheck > 0 that's for updating quantity
$qtyToCheck < 0 I believe that's an exception,

Reply:
gotcha. if we have, say, 2 units of some item in cart, we can currently only remove both (all) of them via the update cart request. not decreasing the quantity to 1. correct?

Reply:
@danac-gs 
1. Under current logic, the update-cart does not allow empty the cart, so if you try to remove the last item, the Bolt modal would directly block that behavior.
2. We can decrease the quantity to 1.
3. If you set the quantity of item to 0 on the Bolt modal, it actually remove the item.

Reply:
@dumega Question here: if we have 3 unit of the same item in cart and going to decrease it to 2. Should we check inventory in this case or not? Going to merge this PR, we can discuss this case separately.

Reply:
@vitaliyreznikov we only ignore the quantity checking if $qtyToCheck == 0, cause that means the customer try to remove the item from cart, then for other cases, we would always process validation.